config/configauth/configauth.go

```diff
@@ -0,0 +1,49 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package configauth
+
+// Authentication defines the auth settings for the receiver
+type Authentication struct {
+	// The attribute (header name) to look for auth data. Optional, default value: "authentication".
+	Attribute string `mapstructure:"attribute"`
```

Comment:
Header?

Reply:
While it does currently use only headers, it might not always be that in the future. Perhaps I should just change the doc to say "... (such as header name)"?

Reply:
`Attribute` is a span concept, and worried that this may be confusing. Please consider to open a separate PR to revisit this, I will merge this initial PR.

Reply:
Created https://github.com/open-telemetry/opentelemetry-collector/issues/1824 to track this.